scene/main/viewport.cpp

```diff
@@ -1955,6 +1957,31 @@ void Viewport::_gui_input_event(Ref<InputEvent> p_event) {
 				gui.drag_data=Variant(); //always clear
 			}*/
 
+			// In case the mouse was released after for example dragging a scrollbar,
+			// check whether the current control is different from the stored one. If
+			// it is different, rather than wait for it to be updated the next time the
+			// mouse is moved, notify the control so that it can e.g. drop the highlight.
+			// This code is duplicated from the mm.is_valid()-case further below.
+			if (gui.mouse_focus) {
+				over = gui.mouse_focus;
+			} else {
+				over = _gui_find_control(mpos);
+			}
+
+			if (over != gui.mouse_over) {
```

Comment:
I have the feeling this should happen only when gui.mouse_focus_mask == 0 , meaning no mouse button is being pressed, otherwise the logic dictates that the events are captured by the control you are dragging from.

Reply:
>
>I have the feeling this should happen only when gui.mouse_focus_mask == 0

This way? Still seems to work for me with the extra condition added. Is it to account for the case where multiple buttons might be pressed and one was released?

 @RandomShaper :
>
> If anything, please add a counterpart comment to the original of the duplicated code about the duplication, so future maintainers know that those pieces of behavior must match. 

Sorry, I missed this comment earlier I added a comment above those sections too.

Reply:
Haha forgot to push.